### PR TITLE
feat(labels): support single choice

### PR DESCRIPTION
**Pull request by bbbmj**
support single choice for labels
/cc @ddysher @supereagle @xiaoq17 
/release-note-none

### CODE REVIEW

**Comment by bbbmj** [comment]
Fix #https://github.com/caicloud/tools/issues/187

File: prow/plugins/label/label.go [hunk 1]
```diff
@@ -119,6 +119,15 @@ func handle(gc githubClient, log *logrus.Entry, e *github.GenericCommentEvent) e
 			continue
 		}
 
+		for _, label := range labels {
+			labelPrefix := strings.Split(labelToAdd, "/")
+			if strings.HasPrefix(label.Name, labelPrefix[0]) {
+				if err := gc.RemoveLabel(org, repo, e.Number, label.Name); err != nil {
```
Comment (deyuanix) [inline prow/plugins/label/label.go]: please only do this for queue and priority.

Comment (bbbmj) [reply prow/plugins/label/label.go]: done  PTAL @ddysher

File: prow/plugins/label/label.go [hunk 2]
```diff
@@ -119,6 +121,18 @@ func handle(gc githubClient, log *logrus.Entry, e *github.GenericCommentEvent) e
 			continue
 		}
 
+		for _, label := range labels {
+			prefixes := strings.Split(*singleChoice, ",")
+			for _, prefix := range prefixes {
```
Comment (deyuanix) [inline prow/plugins/label/label.go]: This works but a better implementation is to make `prefixes` a map and loop on existing labels, remove outdated label if you found a match. Once found, you can break out the loop.

File: prow/plugins/label/label.go [hunk 3]
```diff
@@ -33,6 +34,7 @@ const pluginName = "label"
 var (
 	labelRegex              = regexp.MustCompile(`(?m)^/(area|committee|kind|priority|sig|branch|queue|version)\s*(.*)$`)
 	removeLabelRegex        = regexp.MustCompile(`(?m)^/remove-(area|committee|kind|priority|sig|branch|queue|version)\s*(.*)$`)
+	singleChoice            = flag.String("single-choice", "priority,queue", "Comma separated list of command that needs support single-choice")
```
Comment (deyuanix) [inline prow/plugins/label/label.go]: thinking further, i think `scrum` is also single choice? @xiaoq17

File: prow/plugins/label/label.go (hunk 2 above)
Comment (bbbmj) [reply prow/plugins/label/label.go]: I'm confusing...like this?
```
for _, label := range labels {
	prefixes := strings.Split(*singleChoice, ",")
	prefixesMap := map[string]string{}
	for _, p := range prefixes {
		prefixesMap[strings.ToLower(p)] = p
	}
	labelPrefix := strings.Split(labelToAdd, "/")
	if _, ok := prefixesMap[labelPrefix[0]]; !ok && strings.HasPrefix(label.Name, labelPrefix[0]) {
		if err := gc.RemoveLabel(org, repo, e.Number, label.Name); err != nil {
			log.WithError(err).Errorf("Github failed to remove the following label: %s", label.Name)
		}
		break
	}
}
```

Comment (deyuanix) [reply prow/plugins/label/label.go]: move the `prefixesMap` construction loop to outside of the top-level for lop

**Comment by bbbmj** [comment]
done PTAL @ddysher

**Comment by deyuanix** [comment]
LGTM

**Pull request merged**